internal/pkg/agent/transpiler/utils.go

```diff
@@ -85,6 +103,29 @@ type varIDMap struct {
 	d  *Dict
 }
 
+func getStreams(dict *Dict) *List {
+	node, ok := dict.Find(streamsKey)
+	if !ok {
+		return nil
+	}
+	key, ok := node.(*Key)
+	if !ok {
+		return nil
+	}
+	if key.value == nil {
+		return nil
+	}
+	list, ok := key.value.(*List)
+	if !ok {
+		return nil
+	}
+	if len(list.value) == 0 {
+		// didn't have any streams defined in the list (so no removal should be done)
```

Comment:
nit: This comment should probably be on one of the calls to this function in RenderInputs(), you can't know whether streams were or are removed in this function itself.

Reply:
It is above on line 54. This is just a different context on why when the list is empty that it returns nil instead of the list.